feat header #98

Anouarab2 opened the pull request:
## What does this change?

Resolves issue #98 

<!-- A PR should have enough detail to be understandable far in the future. e.g what is the problem/why is the change needed, how does it solve it and any questions or points of discussion. Prefer copying information from a GitHub issue over linking to it; the card may not always exist and reviewers may not have access to the board. -->

We didnt have a header yet so I looked at the design and made the entire header and the few items like the search bar underneath. i made them responsive already and i used a component in de lib file so we can reuse the header on every page of the website. 

i also noticed that the colors in the custom properties weren't working properly, i tried removing the comma's and that fixed it, so i removed the comma's everywhere

## How Has This Been Tested?
<!-- Link to test results in the Wiki-->

- [ ] [User test]()
- [ ] [Accessibility test]()
- [ ] [Performance test]()
- [x] [Responsive Design test]()
- [x] [Device test]()
- [x] [Browser test]()

## Images

here you can see the website tested on desktop size in safari

<img src="https://github.com/user-attachments/assets/3323fd46-7a38-485b-8723-85ba30e2b149" width="600">

i also tested it on tablet size in firefox

<img src="https://github.com/user-attachments/assets/3ec93755-9df7-4866-bb0a-60982859adc4" width="600">

and heres a mobile test in chrome

<img src="https://github.com/user-attachments/assets/f4eae2ea-6114-422c-806f-bbb6368d8bde" width="200">


<!-- Usually only applicable to UI changes, what did it look like before and what will it look like after? -->


## How to review

<!-- Provide instructions to help others verify the change. This could take the form of "On PROD, do X and witness Y. On this branch, do X and witness Z. " -->

you can review by checking my commit and tell me if i should do something differently.

ColindeGroot commented:
@Anouarab2 
The following could use some fixxing:

submit button is too big and doesn't seem to fit in the house style

Text overlap on the icon in the input;
<img width="837" height="315" alt="image" src="https://github.com/user-attachments/assets/5dc3ef92-8333-446a-a6cf-4cc302c1a3bf" />

Also the title could use a bit  more padding on the sides on mobile: 
<img width="470" height="761" alt="image" src="https://github.com/user-attachments/assets/37f25244-ef08-49e5-baaa-4ff84b04692d" />

Seems fine on keyboard test. Did you conduct a screen reader test etc? (accessibility test)

Anouarab2 commented:
feedback processed i've added padding on the h1, changed the styling on the submit button and fixed the overlap on the text in search bar

here you can see the h1 looks normal now on mobile 
![C86FE4B1-C0EC-4C8D-AB93-AA1714C6E182_1_201_a](https://github.com/user-attachments/assets/381355d8-5f14-4575-82d5-287b85c51036)


and here you can see the search bar having the correct styling without parts overlapping and the submit button fitting the style (no border and custom property color).
![AC32D427-8BC7-4607-9C97-F8A665094B8E_1_105_c](https://github.com/user-attachments/assets/32bcde62-e302-42cf-9cec-1855ba5cf0c6)

Anouarab2 commented:
you also gave me some feedback on the whitespace between the buttons and the searchbar so i will fix that too

Anouarab2 commented:
now the submit button is placed more naturally and theres a clearer distinction between the search/chatbar and the filter button
![491A4FAB-A482-4E00-9745-2BE65583CC63_1_201_a](https://github.com/user-attachments/assets/44c47ab7-831a-4e51-9f83-3ebbefb13991)
